layer4/matchers.go

```diff
@@ -125,10 +125,15 @@ func (MatchRemoteIP) CaddyModule() caddy.ModuleInfo {
 }
 
 // Provision parses m's IP ranges, either from IP or CIDR expressions.
-func (m *MatchRemoteIP) Provision(_ caddy.Context) (err error) {
-	m.cidrs, err = ParseNetworks(m.Ranges)
-	if err != nil {
-		return err
+func (m *MatchRemoteIP) Provision(_ caddy.Context) error {
+	repl := caddy.NewReplacer()
+	for _, addrOrCIDR := range m.Ranges {
+		addrOrCIDR = repl.ReplaceAll(addrOrCIDR, "")
+		prefix, err := caddyhttp.CIDRExpressionToPrefix(addrOrCIDR)
```

Comment:
I wonder if this function should move to `caddy` instead of `caddyhttp` (since it can be used for more than just HTTP).

Reply:
I missed the PR back at the caddy repo but I still wonder about this. What's the right package for this, do we think?